Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -38,21 +38,31 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
             }
 
             // Pawns can take pieces diagonally
-            var diagonalLeft = new Square(location.Row + direction*1, location.Col - 1);
-            if (diagonalLeft.Col >= 0 && board.SquareIsOccupied(diagonalLeft) &&
-                board.GetPiece(diagonalLeft).Player != Player)
+            var diagonalLeft = GetDiagonalMoves(board, -1);
+            if (diagonalLeft != null) availableLocations.Add(diagonalLeft.Value);
+
+
+            var diagonalRight = GetDiagonalMoves(board, 1);
+            if (diagonalRight != null) availableLocations.Add(diagonalRight.Value);
+
+            return availableLocations;
+        }
+
+        private Square? GetDiagonalMoves(Board board, int horizontalDirection)
```

Comment:
Should probably be `GetDiagonalMove` since it's only returning one square? 🙂 

Also, I'd recommend using an enum for the horizontal direction, to enforce that it should only have value 1 or -1 🙂

Reply:
I hadn't thought of using an enum, thanks! I've put those in 046c074 🙂